kahuna/public/js/search/results.js

```diff
@@ -136,9 +145,15 @@ results.controller('SearchResultsCtrl', [
             ctrl.images = [];
 
             // imagesAll will be a sparse array of all the results
+            const totalLength = Math.min(images.total, ctrl.maxResults);
             ctrl.imagesAll = [];
             ctrl.imagesAll.length = Math.min(images.total, ctrl.maxResults);
 
+            // TODO: ultimately we want to manage the state in the
+            // results stream exclusively
+            results.clear();
+            results.resize(totalLength);
```

Comment:
It's the equivalent of [this line](https://github.com/guardian/grid/pull/1200/files#diff-b8c6d4770b5b04294cc8c88d27ae65d3R150). We make it a sparse array with the total length.